src/transformers/models/idefics/image_processing_idefics.py

```diff
@@ -0,0 +1,163 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image processor class for Idefics."""
+
+from typing import Dict, List, Optional, Union
+
+from ...image_processing_utils import BaseImageProcessor
+from ...image_utils import (
+    IMAGENET_STANDARD_MEAN,
+    IMAGENET_STANDARD_STD,
+    ChannelDimension,
+    ImageInput,
+    make_list_of_images,
+    valid_images,
+)
+from ...utils import TensorType, is_torchvision_available, is_vision_available, logging
+
+
+if is_torchvision_available():
+    import torchvision.transforms as transforms
+
+if is_vision_available():
+    import PIL
+    from PIL import Image
```

Comment:
This doesn't need protection as the image processor should only be imported when `is_vision_available()` (protection happens in the inits).